packages/website/src/models/useDateOffset.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+import { getDate } from './instance';
+
+function average(arr: number[]): number {
+  return arr.reduce((a, b) => (a != null ? a + b : b), 0) / arr.filter((n) => n).length;
```

Comment:
This seems a little confusing `a != null` doesn't seem like it would be possible if it gets initialized to `0` (second argument).

I think to handle the base case of `arr.length === 0`, we should add an explicit if function before this reduce to avoid divide by zero to be more clear - maybe something like this:
```js
if(arr.length === 0) return DEFAULT_OFFSET;
const sum = arr.reduce((a, b) => a + b, 0)
return sum / arr.length
```

Reply:
Yeah sorry I meant b != null, the value being passed in, since offsets initializes to an array of 10 empty numbers. arr.length will always be 10, so arr.length === 0 won't work.